Build failure: electrum-ltc

## Steps To Reproduce

`nix-build . -A electrum-ltc`

## Build log

<!-- insert build log in code block in collapsable section -->

<details>

<summary>Build Log</summary>

```
** log edited to not exceed Github comment length limit**

Running phase: pytestCheckPhase
Executing pytestCheckPhase
============================= test session starts ==============================
platform linux -- Python 3.12.7, pytest-8.3.3, pluggy-1.5.0
rootdir: /build/Electrum-LTC-4.2.2.1
collected 0 items / 25 errors                                                  

==================================== ERRORS ====================================
...
_____________ ERROR collecting electrum_ltc/tests/test_lnrouter.py _____________
/nix/store/zv1kaq7f1q20x62kbjv6pfjygw5jmwl6-python3-3.12.7/lib/python3.12/importlib/__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1387: in _gcd_import
    ???
<frozen importlib._bootstrap>:1360: in _find_and_load
    ???
<frozen importlib._bootstrap>:1310: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:488: in _call_with_frames_removed
    ???
<frozen importlib._bootstrap>:1387: in _gcd_import
    ???
<frozen importlib._bootstrap>:1360: in _find_and_load
    ???
<frozen importlib._bootstrap>:1310: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:488: in _call_with_frames_removed
    ???
<frozen importlib._bootstrap>:1387: in _gcd_import
    ???
<frozen importlib._bootstrap>:1360: in _find_and_load
    ???
<frozen importlib._bootstrap>:1331: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:935: in _load_unlocked
    ???
<frozen importlib._bootstrap_external>:995: in exec_module
    ???
<frozen importlib._bootstrap>:488: in _call_with_frames_removed
    ???
electrum_ltc/__init__.py:16: in <module>
    from .wallet import Wallet
electrum_ltc/wallet.py:68: in <module>
    from .storage import StorageEncryptionVersion, WalletStorage
electrum_ltc/storage.py:37: in <module>
    from .wallet_db import WalletDB
electrum_ltc/wallet_db.py:45: in <module>
    from .paymentrequest import PaymentRequest
electrum_ltc/paymentrequest.py:37: in <module>
    from . import paymentrequest_pb2 as pb2
electrum_ltc/paymentrequest_pb2.py:36: in <module>
    _descriptor.FieldDescriptor(
/nix/store/gb9jkhq47nx31pfdr2ppkv7b9r85mg3r-python3.12-protobuf-5.28.3/lib/python3.12/site-packages/google/protobuf/descriptor.py:621: in __new__
    _message.Message._CheckCalledFromGeneratedFile()
E   TypeError: Descriptors cannot be created directly.
E   If this call came from a _pb2.py file, your generated code is out of date and must be regenerated with protoc >= 3.19.0.
E   If you cannot immediately regenerate your protos, some other possible workarounds are:
E    1. Downgrade the protobuf package to 3.20.x or lower.
E    2. Set PROTOCOL_BUFFERS_PYTHON_IMPLEMENTATION=python (but this will use pure-Python parsing and will be much slower).
E   
E   More information: https://developers.google.com/protocol-buffers/docs/news/2022-05-06#python-updates
___________ ERROR collecting electrum_ltc/tests/test_lntransport.py ____________
/nix/store/zv1kaq7f1q20x62kbjv6pfjygw5jmwl6-python3-3.12.7/lib/python3.12/importlib/__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1387: in _gcd_import
    ???
<frozen importlib._bootstrap>:1360: in _find_and_load
    ???
<frozen importlib._bootstrap>:1310: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:488: in _call_with_frames_removed
    ???
<frozen importlib._bootstrap>:1387: in _gcd_import
    ???
<frozen importlib._bootstrap>:1360: in _find_and_load
    ???
<frozen importlib._bootstrap>:1310: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:488: in _call_with_frames_removed
    ???
<frozen importlib._bootstrap>:1387: in _gcd_import
    ???
<frozen importlib._bootstrap>:1360: in _find_and_load
    ???
<frozen importlib._bootstrap>:1331: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:935: in _load_unlocked
    ???
<frozen importlib._bootstrap_external>:995: in exec_module
    ???
<frozen importlib._bootstrap>:488: in _call_with_frames_removed
    ???
electrum_ltc/__init__.py:16: in <module>
    from .wallet import Wallet
electrum_ltc/wallet.py:68: in <module>
    from .storage import StorageEncryptionVersion, WalletStorage
electrum_ltc/storage.py:37: in <module>
    from .wallet_db import WalletDB
electrum_ltc/wallet_db.py:45: in <module>
    from .paymentrequest import PaymentRequest
electrum_ltc/paymentrequest.py:37: in <module>
    from . import paymentrequest_pb2 as pb2
electrum_ltc/paymentrequest_pb2.py:36: in <module>
    _descriptor.FieldDescriptor(
/nix/store/gb9jkhq47nx31pfdr2ppkv7b9r85mg3r-python3.12-protobuf-5.28.3/lib/python3.12/site-packages/google/protobuf/descriptor.py:621: in __new__
    _message.Message._CheckCalledFromGeneratedFile()
E   TypeError: Descriptors cannot be created directly.
E   If this call came from a _pb2.py file, your generated code is out of date and must be regenerated with protoc >= 3.19.0.
E   If you cannot immediately regenerate your protos, some other possible workarounds are:
E    1. Downgrade the protobuf package to 3.20.x or lower.
E    2. Set PROTOCOL_BUFFERS_PYTHON_IMPLEMENTATION=python (but this will use pure-Python parsing and will be much slower).
E   
E   More information: https://developers.google.com/protocol-buffers/docs/news/2022-05-06#python-updates
______________ ERROR collecting electrum_ltc/tests/test_lnutil.py ______________
/nix/store/zv1kaq7f1q20x62kbjv6pfjygw5jmwl6-python3-3.12.7/lib/python3.12/importlib/__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1387: in _gcd_import
    ???
<frozen importlib._bootstrap>:1360: in _find_and_load
    ???
<frozen importlib._bootstrap>:1310: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:488: in _call_with_frames_removed
    ???
<frozen importlib._bootstrap>:1387: in _gcd_import
    ???
<frozen importlib._bootstrap>:1360: in _find_and_load
    ???
<frozen importlib._bootstrap>:1310: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:488: in _call_with_frames_removed
    ???
<frozen importlib._bootstrap>:1387: in _gcd_import
    ???
<frozen importlib._bootstrap>:1360: in _find_and_load
    ???
<frozen importlib._bootstrap>:1331: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:935: in _load_unlocked
    ???
<frozen importlib._bootstrap_external>:995: in exec_module
    ???
<frozen importlib._bootstrap>:488: in _call_with_frames_removed
    ???
electrum_ltc/__init__.py:16: in <module>
    from .wallet import Wallet
electrum_ltc/wallet.py:68: in <module>
    from .storage import StorageEncryptionVersion, WalletStorage
electrum_ltc/storage.py:37: in <module>
    from .wallet_db import WalletDB
electrum_ltc/wallet_db.py:45: in <module>
    from .paymentrequest import PaymentRequest
electrum_ltc/paymentrequest.py:37: in <module>
    from . import paymentrequest_pb2 as pb2
electrum_ltc/paymentrequest_pb2.py:36: in <module>
    _descriptor.FieldDescriptor(
/nix/store/gb9jkhq47nx31pfdr2ppkv7b9r85mg3r-python3.12-protobuf-5.28.3/lib/python3.12/site-packages/google/protobuf/descriptor.py:621: in __new__
    _message.Message._CheckCalledFromGeneratedFile()
E   TypeError: Descriptors cannot be created directly.
E   If this call came from a _pb2.py file, your generated code is out of date and must be regenerated with protoc >= 3.19.0.
E   If you cannot immediately regenerate your protos, some other possible workarounds are:
E    1. Downgrade the protobuf package to 3.20.x or lower.
E    2. Set PROTOCOL_BUFFERS_PYTHON_IMPLEMENTATION=python (but this will use pure-Python parsing and will be much slower).
E   
E   More information: https://developers.google.com/protocol-buffers/docs/news/2022-05-06#python-updates
_____________ ERROR collecting electrum_ltc/tests/test_mnemonic.py _____________
/nix/store/zv1kaq7f1q20x62kbjv6pfjygw5jmwl6-python3-3.12.7/lib/python3.12/importlib/__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1387: in _gcd_import
    ???
<frozen importlib._bootstrap>:1360: in _find_and_load
    ???
<frozen importlib._bootstrap>:1310: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:488: in _call_with_frames_removed
    ???
<frozen importlib._bootstrap>:1387: in _gcd_import
    ???
<frozen importlib._bootstrap>:1360: in _find_and_load
    ???
<frozen importlib._bootstrap>:1310: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:488: in _call_with_frames_removed
    ???
<frozen importlib._bootstrap>:1387: in _gcd_import
    ???
<frozen importlib._bootstrap>:1360: in _find_and_load
    ???
<frozen importlib._bootstrap>:1331: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:935: in _load_unlocked
    ???
<frozen importlib._bootstrap_external>:995: in exec_module
    ???
<frozen importlib._bootstrap>:488: in _call_with_frames_removed
    ???
electrum_ltc/__init__.py:16: in <module>
    from .wallet import Wallet
electrum_ltc/wallet.py:68: in <module>
    from .storage import StorageEncryptionVersion, WalletStorage
electrum_ltc/storage.py:37: in <module>
    from .wallet_db import WalletDB
electrum_ltc/wallet_db.py:45: in <module>
    from .paymentrequest import PaymentRequest
electrum_ltc/paymentrequest.py:37: in <module>
    from . import paymentrequest_pb2 as pb2
electrum_ltc/paymentrequest_pb2.py:36: in <module>
    _descriptor.FieldDescriptor(
/nix/store/gb9jkhq47nx31pfdr2ppkv7b9r85mg3r-python3.12-protobuf-5.28.3/lib/python3.12/site-packages/google/protobuf/descriptor.py:621: in __new__
    _message.Message._CheckCalledFromGeneratedFile()
E   TypeError: Descriptors cannot be created directly.
E   If this call came from a _pb2.py file, your generated code is out of date and must be regenerated with protoc >= 3.19.0.
E   If you cannot immediately regenerate your protos, some other possible workarounds are:
E    1. Downgrade the protobuf package to 3.20.x or lower.
E    2. Set PROTOCOL_BUFFERS_PYTHON_IMPLEMENTATION=python (but this will use pure-Python parsing and will be much slower).
E   
E   More information: https://developers.google.com/protocol-buffers/docs/news/2022-05-06#python-updates
____________ ERROR collecting electrum_ltc/tests/test_mpp_split.py _____________
/nix/store/zv1kaq7f1q20x62kbjv6pfjygw5jmwl6-python3-3.12.7/lib/python3.12/importlib/__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1387: in _gcd_import
    ???
<frozen importlib._bootstrap>:1360: in _find_and_load
    ???
<frozen importlib._bootstrap>:1310: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:488: in _call_with_frames_removed
    ???
<frozen importlib._bootstrap>:1387: in _gcd_import
    ???
<frozen importlib._bootstrap>:1360: in _find_and_load
    ???
<frozen importlib._bootstrap>:1310: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:488: in _call_with_frames_removed
    ???
<frozen importlib._bootstrap>:1387: in _gcd_import
    ???
<frozen importlib._bootstrap>:1360: in _find_and_load
    ???
<frozen importlib._bootstrap>:1331: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:935: in _load_unlocked
    ???
<frozen importlib._bootstrap_external>:995: in exec_module
    ???
<frozen importlib._bootstrap>:488: in _call_with_frames_removed
    ???
electrum_ltc/__init__.py:16: in <module>
    from .wallet import Wallet
electrum_ltc/wallet.py:68: in <module>
    from .storage import StorageEncryptionVersion, WalletStorage
electrum_ltc/storage.py:37: in <module>
    from .wallet_db import WalletDB
electrum_ltc/wallet_db.py:45: in <module>
    from .paymentrequest import PaymentRequest
electrum_ltc/paymentrequest.py:37: in <module>
    from . import paymentrequest_pb2 as pb2
electrum_ltc/paymentrequest_pb2.py:36: in <module>
    _descriptor.FieldDescriptor(
/nix/store/gb9jkhq47nx31pfdr2ppkv7b9r85mg3r-python3.12-protobuf-5.28.3/lib/python3.12/site-packages/google/protobuf/descriptor.py:621: in __new__
    _message.Message._CheckCalledFromGeneratedFile()
E   TypeError: Descriptors cannot be created directly.
E   If this call came from a _pb2.py file, your generated code is out of date and must be regenerated with protoc >= 3.19.0.
E   If you cannot immediately regenerate your protos, some other possible workarounds are:
E    1. Downgrade the protobuf package to 3.20.x or lower.
E    2. Set PROTOCOL_BUFFERS_PYTHON_IMPLEMENTATION=python (but this will use pure-Python parsing and will be much slower).
E   
E   More information: https://developers.google.com/protocol-buffers/docs/news/2022-05-06#python-updates
_____________ ERROR collecting electrum_ltc/tests/test_network.py ______________
/nix/store/zv1kaq7f1q20x62kbjv6pfjygw5jmwl6-python3-3.12.7/lib/python3.12/importlib/__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1387: in _gcd_import
    ???
<frozen importlib._bootstrap>:1360: in _find_and_load
    ???
<frozen importlib._bootstrap>:1310: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:488: in _call_with_frames_removed
    ???
<frozen importlib._bootstrap>:1387: in _gcd_import
    ???
<frozen importlib._bootstrap>:1360: in _find_and_load
    ???
<frozen importlib._bootstrap>:1310: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:488: in _call_with_frames_removed
    ???
<frozen importlib._bootstrap>:1387: in _gcd_import
    ???
<frozen importlib._bootstrap>:1360: in _find_and_load
    ???
<frozen importlib._bootstrap>:1331: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:935: in _load_unlocked
    ???
<frozen importlib._bootstrap_external>:995: in exec_module
    ???
<frozen importlib._bootstrap>:488: in _call_with_frames_removed
    ???
electrum_ltc/__init__.py:16: in <module>
    from .wallet import Wallet
electrum_ltc/wallet.py:68: in <module>
    from .storage import StorageEncryptionVersion, WalletStorage
electrum_ltc/storage.py:37: in <module>
    from .wallet_db import WalletDB
electrum_ltc/wallet_db.py:45: in <module>
    from .paymentrequest import PaymentRequest
electrum_ltc/paymentrequest.py:37: in <module>
    from . import paymentrequest_pb2 as pb2
electrum_ltc/paymentrequest_pb2.py:36: in <module>
    _descriptor.FieldDescriptor(
/nix/store/gb9jkhq47nx31pfdr2ppkv7b9r85mg3r-python3.12-protobuf-5.28.3/lib/python3.12/site-packages/google/protobuf/descriptor.py:621: in __new__
    _message.Message._CheckCalledFromGeneratedFile()
E   TypeError: Descriptors cannot be created directly.
E   If this call came from a _pb2.py file, your generated code is out of date and must be regenerated with protoc >= 3.19.0.
E   If you cannot immediately regenerate your protos, some other possible workarounds are:
E    1. Downgrade the protobuf package to 3.20.x or lower.
E    2. Set PROTOCOL_BUFFERS_PYTHON_IMPLEMENTATION=python (but this will use pure-Python parsing and will be much slower).
E   
E   More information: https://developers.google.com/protocol-buffers/docs/news/2022-05-06#python-updates
_______________ ERROR collecting electrum_ltc/tests/test_psbt.py _______________
/nix/store/zv1kaq7f1q20x62kbjv6pfjygw5jmwl6-python3-3.12.7/lib/python3.12/importlib/__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1387: in _gcd_import
    ???
<frozen importlib._bootstrap>:1360: in _find_and_load
    ???
<frozen importlib._bootstrap>:1310: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:488: in _call_with_frames_removed
    ???
<frozen importlib._bootstrap>:1387: in _gcd_import
    ???
<frozen importlib._bootstrap>:1360: in _find_and_load
    ???
<frozen importlib._bootstrap>:1310: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:488: in _call_with_frames_removed
    ???
<frozen importlib._bootstrap>:1387: in _gcd_import
    ???
<frozen importlib._bootstrap>:1360: in _find_and_load
    ???
<frozen importlib._bootstrap>:1331: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:935: in _load_unlocked
    ???
<frozen importlib._bootstrap_external>:995: in exec_module
    ???
<frozen importlib._bootstrap>:488: in _call_with_frames_removed
    ???
electrum_ltc/__init__.py:16: in <module>
    from .wallet import Wallet
electrum_ltc/wallet.py:68: in <module>
    from .storage import StorageEncryptionVersion, WalletStorage
electrum_ltc/storage.py:37: in <module>
    from .wallet_db import WalletDB
electrum_ltc/wallet_db.py:45: in <module>
    from .paymentrequest import PaymentRequest
electrum_ltc/paymentrequest.py:37: in <module>
    from . import paymentrequest_pb2 as pb2
electrum_ltc/paymentrequest_pb2.py:36: in <module>
    _descriptor.FieldDescriptor(
/nix/store/gb9jkhq47nx31pfdr2ppkv7b9r85mg3r-python3.12-protobuf-5.28.3/lib/python3.12/site-packages/google/protobuf/descriptor.py:621: in __new__
    _message.Message._CheckCalledFromGeneratedFile()
E   TypeError: Descriptors cannot be created directly.
E   If this call came from a _pb2.py file, your generated code is out of date and must be regenerated with protoc >= 3.19.0.
E   If you cannot immediately regenerate your protos, some other possible workarounds are:
E    1. Downgrade the protobuf package to 3.20.x or lower.
E    2. Set PROTOCOL_BUFFERS_PYTHON_IMPLEMENTATION=python (but this will use pure-Python parsing and will be much slower).
E   
E   More information: https://developers.google.com/protocol-buffers/docs/news/2022-05-06#python-updates
_____________ ERROR collecting electrum_ltc/tests/test_revealer.py _____________
/nix/store/zv1kaq7f1q20x62kbjv6pfjygw5jmwl6-python3-3.12.7/lib/python3.12/importlib/__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1387: in _gcd_import
    ???
<frozen importlib._bootstrap>:1360: in _find_and_load
    ???
<frozen importlib._bootstrap>:1310: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:488: in _call_with_frames_removed
    ???
<frozen importlib._bootstrap>:1387: in _gcd_import
    ???
<frozen importlib._bootstrap>:1360: in _find_and_load
    ???
<frozen importlib._bootstrap>:1310: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:488: in _call_with_frames_removed
    ???
<frozen importlib._bootstrap>:1387: in _gcd_import
    ???
<frozen importlib._bootstrap>:1360: in _find_and_load
    ???
<frozen importlib._bootstrap>:1331: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:935: in _load_unlocked
    ???
<frozen importlib._bootstrap_external>:995: in exec_module
    ???
<frozen importlib._bootstrap>:488: in _call_with_frames_removed
    ???
electrum_ltc/__init__.py:16: in <module>
    from .wallet import Wallet
electrum_ltc/wallet.py:68: in <module>
    from .storage import StorageEncryptionVersion, WalletStorage
electrum_ltc/storage.py:37: in <module>
    from .wallet_db import WalletDB
electrum_ltc/wallet_db.py:45: in <module>
    from .paymentrequest import PaymentRequest
electrum_ltc/paymentrequest.py:37: in <module>
    from . import paymentrequest_pb2 as pb2
electrum_ltc/paymentrequest_pb2.py:36: in <module>
    _descriptor.FieldDescriptor(
/nix/store/gb9jkhq47nx31pfdr2ppkv7b9r85mg3r-python3.12-protobuf-5.28.3/lib/python3.12/site-packages/google/protobuf/descriptor.py:621: in __new__
    _message.Message._CheckCalledFromGeneratedFile()
E   TypeError: Descriptors cannot be created directly.
E   If this call came from a _pb2.py file, your generated code is out of date and must be regenerated with protoc >= 3.19.0.
E   If you cannot immediately regenerate your protos, some other possible workarounds are:
E    1. Downgrade the protobuf package to 3.20.x or lower.
E    2. Set PROTOCOL_BUFFERS_PYTHON_IMPLEMENTATION=python (but this will use pure-Python parsing and will be much slower).
E   
E   More information: https://developers.google.com/protocol-buffers/docs/news/2022-05-06#python-updates
__________ ERROR collecting electrum_ltc/tests/test_simple_config.py ___________
/nix/store/zv1kaq7f1q20x62kbjv6pfjygw5jmwl6-python3-3.12.7/lib/python3.12/importlib/__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1387: in _gcd_import
    ???
<frozen importlib._bootstrap>:1360: in _find_and_load
    ???
<frozen importlib._bootstrap>:1310: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:488: in _call_with_frames_removed
    ???
<frozen importlib._bootstrap>:1387: in _gcd_import
    ???
<frozen importlib._bootstrap>:1360: in _find_and_load
    ???
<frozen importlib._bootstrap>:1310: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:488: in _call_with_frames_removed
    ???
<frozen importlib._bootstrap>:1387: in _gcd_import
    ???
<frozen importlib._bootstrap>:1360: in _find_and_load
    ???
<frozen importlib._bootstrap>:1331: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:935: in _load_unlocked
    ???
<frozen importlib._bootstrap_external>:995: in exec_module
    ???
<frozen importlib._bootstrap>:488: in _call_with_frames_removed
    ???
electrum_ltc/__init__.py:16: in <module>
    from .wallet import Wallet
electrum_ltc/wallet.py:68: in <module>
    from .storage import StorageEncryptionVersion, WalletStorage
electrum_ltc/storage.py:37: in <module>
    from .wallet_db import WalletDB
electrum_ltc/wallet_db.py:45: in <module>
    from .paymentrequest import PaymentRequest
electrum_ltc/paymentrequest.py:37: in <module>
    from . import paymentrequest_pb2 as pb2
electrum_ltc/paymentrequest_pb2.py:36: in <module>
    _descriptor.FieldDescriptor(
/nix/store/gb9jkhq47nx31pfdr2ppkv7b9r85mg3r-python3.12-protobuf-5.28.3/lib/python3.12/site-packages/google/protobuf/descriptor.py:621: in __new__
    _message.Message._CheckCalledFromGeneratedFile()
E   TypeError: Descriptors cannot be created directly.
E   If this call came from a _pb2.py file, your generated code is out of date and must be regenerated with protoc >= 3.19.0.
E   If you cannot immediately regenerate your protos, some other possible workarounds are:
E    1. Downgrade the protobuf package to 3.20.x or lower.
E    2. Set PROTOCOL_BUFFERS_PYTHON_IMPLEMENTATION=python (but this will use pure-Python parsing and will be much slower).
E   
E   More information: https://developers.google.com/protocol-buffers/docs/news/2022-05-06#python-updates
_________ ERROR collecting electrum_ltc/tests/test_storage_upgrade.py __________
/nix/store/zv1kaq7f1q20x62kbjv6pfjygw5jmwl6-python3-3.12.7/lib/python3.12/importlib/__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1387: in _gcd_import
    ???
<frozen importlib._bootstrap>:1360: in _find_and_load
    ???
<frozen importlib._bootstrap>:1310: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:488: in _call_with_frames_removed
    ???
<frozen importlib._bootstrap>:1387: in _gcd_import
    ???
<frozen importlib._bootstrap>:1360: in _find_and_load
    ???
<frozen importlib._bootstrap>:1310: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:488: in _call_with_frames_removed
    ???
<frozen importlib._bootstrap>:1387: in _gcd_import
    ???
<frozen importlib._bootstrap>:1360: in _find_and_load
    ???
<frozen importlib._bootstrap>:1331: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:935: in _load_unlocked
    ???
<frozen importlib._bootstrap_external>:995: in exec_module
    ???
<frozen importlib._bootstrap>:488: in _call_with_frames_removed
    ???
electrum_ltc/__init__.py:16: in <module>
    from .wallet import Wallet
electrum_ltc/wallet.py:68: in <module>
    from .storage import StorageEncryptionVersion, WalletStorage
electrum_ltc/storage.py:37: in <module>
    from .wallet_db import WalletDB
electrum_ltc/wallet_db.py:45: in <module>
    from .paymentrequest import PaymentRequest
electrum_ltc/paymentrequest.py:37: in <module>
    from . import paymentrequest_pb2 as pb2
electrum_ltc/paymentrequest_pb2.py:36: in <module>
    _descriptor.FieldDescriptor(
/nix/store/gb9jkhq47nx31pfdr2ppkv7b9r85mg3r-python3.12-protobuf-5.28.3/lib/python3.12/site-packages/google/protobuf/descriptor.py:621: in __new__
    _message.Message._CheckCalledFromGeneratedFile()
E   TypeError: Descriptors cannot be created directly.
E   If this call came from a _pb2.py file, your generated code is out of date and must be regenerated with protoc >= 3.19.0.
E   If you cannot immediately regenerate your protos, some other possible workarounds are:
E    1. Downgrade the protobuf package to 3.20.x or lower.
E    2. Set PROTOCOL_BUFFERS_PYTHON_IMPLEMENTATION=python (but this will use pure-Python parsing and will be much slower).
E   
E   More information: https://developers.google.com/protocol-buffers/docs/news/2022-05-06#python-updates
___________ ERROR collecting electrum_ltc/tests/test_transaction.py ____________
/nix/store/zv1kaq7f1q20x62kbjv6pfjygw5jmwl6-python3-3.12.7/lib/python3.12/importlib/__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1387: in _gcd_import
    ???
<frozen importlib._bootstrap>:1360: in _find_and_load
    ???
<frozen importlib._bootstrap>:1310: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:488: in _call_with_frames_removed
    ???
<frozen importlib._bootstrap>:1387: in _gcd_import
    ???
<frozen importlib._bootstrap>:1360: in _find_and_load
    ???
<frozen importlib._bootstrap>:1310: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:488: in _call_with_frames_removed
    ???
<frozen importlib._bootstrap>:1387: in _gcd_import
    ???
<frozen importlib._bootstrap>:1360: in _find_and_load
    ???
<frozen importlib._bootstrap>:1331: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:935: in _load_unlocked
    ???
<frozen importlib._bootstrap_external>:995: in exec_module
    ???
<frozen importlib._bootstrap>:488: in _call_with_frames_removed
    ???
electrum_ltc/__init__.py:16: in <module>
    from .wallet import Wallet
electrum_ltc/wallet.py:68: in <module>
    from .storage import StorageEncryptionVersion, WalletStorage
electrum_ltc/storage.py:37: in <module>
    from .wallet_db import WalletDB
electrum_ltc/wallet_db.py:45: in <module>
    from .paymentrequest import PaymentRequest
electrum_ltc/paymentrequest.py:37: in <module>
    from . import paymentrequest_pb2 as pb2
electrum_ltc/paymentrequest_pb2.py:36: in <module>
    _descriptor.FieldDescriptor(
/nix/store/gb9jkhq47nx31pfdr2ppkv7b9r85mg3r-python3.12-protobuf-5.28.3/lib/python3.12/site-packages/google/protobuf/descriptor.py:621: in __new__
    _message.Message._CheckCalledFromGeneratedFile()
E   TypeError: Descriptors cannot be created directly.
E   If this call came from a _pb2.py file, your generated code is out of date and must be regenerated with protoc >= 3.19.0.
E   If you cannot immediately regenerate your protos, some other possible workarounds are:
E    1. Downgrade the protobuf package to 3.20.x or lower.
E    2. Set PROTOCOL_BUFFERS_PYTHON_IMPLEMENTATION=python (but this will use pure-Python parsing and will be much slower).
E   
E   More information: https://developers.google.com/protocol-buffers/docs/news/2022-05-06#python-updates
_______________ ERROR collecting electrum_ltc/tests/test_util.py _______________
/nix/store/zv1kaq7f1q20x62kbjv6pfjygw5jmwl6-python3-3.12.7/lib/python3.12/importlib/__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1387: in _gcd_import
    ???
<frozen importlib._bootstrap>:1360: in _find_and_load
    ???
<frozen importlib._bootstrap>:1310: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:488: in _call_with_frames_removed
    ???
<frozen importlib._bootstrap>:1387: in _gcd_import
    ???
<frozen importlib._bootstrap>:1360: in _find_and_load
    ???
<frozen importlib._bootstrap>:1310: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:488: in _call_with_frames_removed
    ???
<frozen importlib._bootstrap>:1387: in _gcd_import
    ???
<frozen importlib._bootstrap>:1360: in _find_and_load
    ???
<frozen importlib._bootstrap>:1331: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:935: in _load_unlocked
    ???
<frozen importlib._bootstrap_external>:995: in exec_module
    ???
<frozen importlib._bootstrap>:488: in _call_with_frames_removed
    ???
electrum_ltc/__init__.py:16: in <module>
    from .wallet import Wallet
electrum_ltc/wallet.py:68: in <module>
    from .storage import StorageEncryptionVersion, WalletStorage
electrum_ltc/storage.py:37: in <module>
    from .wallet_db import WalletDB
electrum_ltc/wallet_db.py:45: in <module>
    from .paymentrequest import PaymentRequest
electrum_ltc/paymentrequest.py:37: in <module>
    from . import paymentrequest_pb2 as pb2
electrum_ltc/paymentrequest_pb2.py:36: in <module>
    _descriptor.FieldDescriptor(
/nix/store/gb9jkhq47nx31pfdr2ppkv7b9r85mg3r-python3.12-protobuf-5.28.3/lib/python3.12/site-packages/google/protobuf/descriptor.py:621: in __new__
    _message.Message._CheckCalledFromGeneratedFile()
E   TypeError: Descriptors cannot be created directly.
E   If this call came from a _pb2.py file, your generated code is out of date and must be regenerated with protoc >= 3.19.0.
E   If you cannot immediately regenerate your protos, some other possible workarounds are:
E    1. Downgrade the protobuf package to 3.20.x or lower.
E    2. Set PROTOCOL_BUFFERS_PYTHON_IMPLEMENTATION=python (but this will use pure-Python parsing and will be much slower).
E   
E   More information: https://developers.google.com/protocol-buffers/docs/news/2022-05-06#python-updates
_____________ ERROR collecting electrum_ltc/tests/test_verifier.py _____________
/nix/store/zv1kaq7f1q20x62kbjv6pfjygw5jmwl6-python3-3.12.7/lib/python3.12/importlib/__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1387: in _gcd_import
    ???
<frozen importlib._bootstrap>:1360: in _find_and_load
    ???
<frozen importlib._bootstrap>:1310: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:488: in _call_with_frames_removed
    ???
<frozen importlib._bootstrap>:1387: in _gcd_import
    ???
<frozen importlib._bootstrap>:1360: in _find_and_load
    ???
<frozen importlib._bootstrap>:1310: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:488: in _call_with_frames_removed
    ???
<frozen importlib._bootstrap>:1387: in _gcd_import
    ???
<frozen importlib._bootstrap>:1360: in _find_and_load
    ???
<frozen importlib._bootstrap>:1331: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:935: in _load_unlocked
    ???
<frozen importlib._bootstrap_external>:995: in exec_module
    ???
<frozen importlib._bootstrap>:488: in _call_with_frames_removed
    ???
electrum_ltc/__init__.py:16: in <module>
    from .wallet import Wallet
electrum_ltc/wallet.py:68: in <module>
    from .storage import StorageEncryptionVersion, WalletStorage
electrum_ltc/storage.py:37: in <module>
    from .wallet_db import WalletDB
electrum_ltc/wallet_db.py:45: in <module>
    from .paymentrequest import PaymentRequest
electrum_ltc/paymentrequest.py:37: in <module>
    from . import paymentrequest_pb2 as pb2
electrum_ltc/paymentrequest_pb2.py:36: in <module>
    _descriptor.FieldDescriptor(
/nix/store/gb9jkhq47nx31pfdr2ppkv7b9r85mg3r-python3.12-protobuf-5.28.3/lib/python3.12/site-packages/google/protobuf/descriptor.py:621: in __new__
    _message.Message._CheckCalledFromGeneratedFile()
E   TypeError: Descriptors cannot be created directly.
E   If this call came from a _pb2.py file, your generated code is out of date and must be regenerated with protoc >= 3.19.0.
E   If you cannot immediately regenerate your protos, some other possible workarounds are:
E    1. Downgrade the protobuf package to 3.20.x or lower.
E    2. Set PROTOCOL_BUFFERS_PYTHON_IMPLEMENTATION=python (but this will use pure-Python parsing and will be much slower).
E   
E   More information: https://developers.google.com/protocol-buffers/docs/news/2022-05-06#python-updates
______________ ERROR collecting electrum_ltc/tests/test_wallet.py ______________
/nix/store/zv1kaq7f1q20x62kbjv6pfjygw5jmwl6-python3-3.12.7/lib/python3.12/importlib/__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1387: in _gcd_import
    ???
<frozen importlib._bootstrap>:1360: in _find_and_load
    ???
<frozen importlib._bootstrap>:1310: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:488: in _call_with_frames_removed
    ???
<frozen importlib._bootstrap>:1387: in _gcd_import
    ???
<frozen importlib._bootstrap>:1360: in _find_and_load
    ???
<frozen importlib._bootstrap>:1310: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:488: in _call_with_frames_removed
    ???
<frozen importlib._bootstrap>:1387: in _gcd_import
    ???
<frozen importlib._bootstrap>:1360: in _find_and_load
    ???
<frozen importlib._bootstrap>:1331: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:935: in _load_unlocked
    ???
<frozen importlib._bootstrap_external>:995: in exec_module
    ???
<frozen importlib._bootstrap>:488: in _call_with_frames_removed
    ???
electrum_ltc/__init__.py:16: in <module>
    from .wallet import Wallet
electrum_ltc/wallet.py:68: in <module>
    from .storage import StorageEncryptionVersion, WalletStorage
electrum_ltc/storage.py:37: in <module>
    from .wallet_db import WalletDB
electrum_ltc/wallet_db.py:45: in <module>
    from .paymentrequest import PaymentRequest
electrum_ltc/paymentrequest.py:37: in <module>
    from . import paymentrequest_pb2 as pb2
electrum_ltc/paymentrequest_pb2.py:36: in <module>
    _descriptor.FieldDescriptor(
/nix/store/gb9jkhq47nx31pfdr2ppkv7b9r85mg3r-python3.12-protobuf-5.28.3/lib/python3.12/site-packages/google/protobuf/descriptor.py:621: in __new__
    _message.Message._CheckCalledFromGeneratedFile()
E   TypeError: Descriptors cannot be created directly.
E   If this call came from a _pb2.py file, your generated code is out of date and must be regenerated with protoc >= 3.19.0.
E   If you cannot immediately regenerate your protos, some other possible workarounds are:
E    1. Downgrade the protobuf package to 3.20.x or lower.
E    2. Set PROTOCOL_BUFFERS_PYTHON_IMPLEMENTATION=python (but this will use pure-Python parsing and will be much slower).
E   
E   More information: https://developers.google.com/protocol-buffers/docs/news/2022-05-06#python-updates
_________ ERROR collecting electrum_ltc/tests/test_wallet_vertical.py __________
/nix/store/zv1kaq7f1q20x62kbjv6pfjygw5jmwl6-python3-3.12.7/lib/python3.12/importlib/__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1387: in _gcd_import
    ???
<frozen importlib._bootstrap>:1360: in _find_and_load
    ???
<frozen importlib._bootstrap>:1310: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:488: in _call_with_frames_removed
    ???
<frozen importlib._bootstrap>:1387: in _gcd_import
    ???
<frozen importlib._bootstrap>:1360: in _find_and_load
    ???
<frozen importlib._bootstrap>:1310: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:488: in _call_with_frames_removed
    ???
<frozen importlib._bootstrap>:1387: in _gcd_import
    ???
<frozen importlib._bootstrap>:1360: in _find_and_load
    ???
<frozen importlib._bootstrap>:1331: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:935: in _load_unlocked
    ???
<frozen importlib._bootstrap_external>:995: in exec_module
    ???
<frozen importlib._bootstrap>:488: in _call_with_frames_removed
    ???
electrum_ltc/__init__.py:16: in <module>
    from .wallet import Wallet
electrum_ltc/wallet.py:68: in <module>
    from .storage import StorageEncryptionVersion, WalletStorage
electrum_ltc/storage.py:37: in <module>
    from .wallet_db import WalletDB
electrum_ltc/wallet_db.py:45: in <module>
    from .paymentrequest import PaymentRequest
electrum_ltc/paymentrequest.py:37: in <module>
    from . import paymentrequest_pb2 as pb2
electrum_ltc/paymentrequest_pb2.py:36: in <module>
    _descriptor.FieldDescriptor(
/nix/store/gb9jkhq47nx31pfdr2ppkv7b9r85mg3r-python3.12-protobuf-5.28.3/lib/python3.12/site-packages/google/protobuf/descriptor.py:621: in __new__
    _message.Message._CheckCalledFromGeneratedFile()
E   TypeError: Descriptors cannot be created directly.
E   If this call came from a _pb2.py file, your generated code is out of date and must be regenerated with protoc >= 3.19.0.
E   If you cannot immediately regenerate your protos, some other possible workarounds are:
E    1. Downgrade the protobuf package to 3.20.x or lower.
E    2. Set PROTOCOL_BUFFERS_PYTHON_IMPLEMENTATION=python (but this will use pure-Python parsing and will be much slower).
E   
E   More information: https://developers.google.com/protocol-buffers/docs/news/2022-05-06#python-updates
_______________ ERROR collecting electrum_ltc/tests/test_x509.py _______________
/nix/store/zv1kaq7f1q20x62kbjv6pfjygw5jmwl6-python3-3.12.7/lib/python3.12/importlib/__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1387: in _gcd_import
    ???
<frozen importlib._bootstrap>:1360: in _find_and_load
    ???
<frozen importlib._bootstrap>:1310: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:488: in _call_with_frames_removed
    ???
<frozen importlib._bootstrap>:1387: in _gcd_import
    ???
<frozen importlib._bootstrap>:1360: in _find_and_load
    ???
<frozen importlib._bootstrap>:1310: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:488: in _call_with_frames_removed
    ???
<frozen importlib._bootstrap>:1387: in _gcd_import
    ???
<frozen importlib._bootstrap>:1360: in _find_and_load
    ???
<frozen importlib._bootstrap>:1331: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:935: in _load_unlocked
    ???
<frozen importlib._bootstrap_external>:995: in exec_module
    ???
<frozen importlib._bootstrap>:488: in _call_with_frames_removed
    ???
electrum_ltc/__init__.py:16: in <module>
    from .wallet import Wallet
electrum_ltc/wallet.py:68: in <module>
    from .storage import StorageEncryptionVersion, WalletStorage
electrum_ltc/storage.py:37: in <module>
    from .wallet_db import WalletDB
electrum_ltc/wallet_db.py:45: in <module>
    from .paymentrequest import PaymentRequest
electrum_ltc/paymentrequest.py:37: in <module>
    from . import paymentrequest_pb2 as pb2
electrum_ltc/paymentrequest_pb2.py:36: in <module>
    _descriptor.FieldDescriptor(
/nix/store/gb9jkhq47nx31pfdr2ppkv7b9r85mg3r-python3.12-protobuf-5.28.3/lib/python3.12/site-packages/google/protobuf/descriptor.py:621: in __new__
    _message.Message._CheckCalledFromGeneratedFile()
E   TypeError: Descriptors cannot be created directly.
E   If this call came from a _pb2.py file, your generated code is out of date and must be regenerated with protoc >= 3.19.0.
E   If you cannot immediately regenerate your protos, some other possible workarounds are:
E    1. Downgrade the protobuf package to 3.20.x or lower.
E    2. Set PROTOCOL_BUFFERS_PYTHON_IMPLEMENTATION=python (but this will use pure-Python parsing and will be much slower).
E   
E   More information: https://developers.google.com/protocol-buffers/docs/news/2022-05-06#python-updates
=========================== short test summary info ============================
ERROR electrum_ltc/tests/test_bitcoin.py - TypeError: Descriptors cannot be created directly.
ERROR electrum_ltc/tests/test_blockchain.py - TypeError: Descriptors cannot be created directly.
ERROR electrum_ltc/tests/test_bolt11.py - TypeError: Descriptors cannot be created directly.
ERROR electrum_ltc/tests/test_coinchooser.py - TypeError: Descriptors cannot be created directly.
ERROR electrum_ltc/tests/test_commands.py - TypeError: Descriptors cannot be created directly.
ERROR electrum_ltc/tests/test_lnchannel.py - TypeError: Descriptors cannot be created directly.
ERROR electrum_ltc/tests/test_lnhtlc.py - TypeError: Descriptors cannot be created directly.
ERROR electrum_ltc/tests/test_lnmsg.py - TypeError: Descriptors cannot be created directly.
ERROR electrum_ltc/tests/test_lnpeer.py - TypeError: Descriptors cannot be created directly.
ERROR electrum_ltc/tests/test_lnrouter.py - TypeError: Descriptors cannot be created directly.
ERROR electrum_ltc/tests/test_lntransport.py - TypeError: Descriptors cannot be created directly.
ERROR electrum_ltc/tests/test_lnutil.py - TypeError: Descriptors cannot be created directly.
ERROR electrum_ltc/tests/test_mnemonic.py - TypeError: Descriptors cannot be created directly.
ERROR electrum_ltc/tests/test_mpp_split.py - TypeError: Descriptors cannot be created directly.
ERROR electrum_ltc/tests/test_network.py - TypeError: Descriptors cannot be created directly.
ERROR electrum_ltc/tests/test_psbt.py - TypeError: Descriptors cannot be created directly.
ERROR electrum_ltc/tests/test_revealer.py - TypeError: Descriptors cannot be created directly.
ERROR electrum_ltc/tests/test_simple_config.py - TypeError: Descriptors cannot be created directly.
ERROR electrum_ltc/tests/test_storage_upgrade.py - TypeError: Descriptors cannot be created directly.
ERROR electrum_ltc/tests/test_transaction.py - TypeError: Descriptors cannot be created directly.
ERROR electrum_ltc/tests/test_util.py - TypeError: Descriptors cannot be created directly.
ERROR electrum_ltc/tests/test_verifier.py - TypeError: Descriptors cannot be created directly.
ERROR electrum_ltc/tests/test_wallet.py - TypeError: Descriptors cannot be created directly.
ERROR electrum_ltc/tests/test_wallet_vertical.py - TypeError: Descriptors cannot be created directly.
ERROR electrum_ltc/tests/test_x509.py - TypeError: Descriptors cannot be created directly.
!!!!!!!!!!!!!!!!!!! Interrupted: 25 errors during collection !!!!!!!!!!!!!!!!!!!
============================== 25 errors in 4.07s ==============================
```

</details>

## Additional context

Affects nixpkgs branches `master`, `nixos-24.11`
Works in branch `nixos-24.05`

## Metadata

Platform: `x86_64-linux`

## Notify maintainers

@bbjubjub2494

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/applications/misc/electrum/ltc.nix
+++ b/pkgs/applications/misc/electrum/ltc.nix
@@ -109,6 +109,9 @@ python3.pkgs.buildPythonApplication {
     # copy the patched `/run_electrum` over `/electrum/electrum`
     # so the aiorpcx compatibility patch is used
     cp run_electrum electrum_ltc/electrum-ltc
+
+    # refresh stale generated code, per electrum_ltc/paymentrequest.py line 40
+    protoc --proto_path=electrum_ltc/ --python_out=electrum_ltc/ electrum_ltc/paymentrequest.proto
   '';
 
   preBuild =

